erpnext/selling/page/order_desk/order_desk.js

```diff
@@ -0,0 +1,1466 @@
+/* global Clusterize */
+frappe.provide('erpnext.pos');
+
+frappe.pages['order-desk'].refresh = function(wrapper) {
+	if (wrapper.pos) {
+		wrapper.pos.make_new_order();
+	}
+}
+
+frappe.pages['order-desk'].on_page_load = function(wrapper) {
+	frappe.ui.make_app_page({
+		parent: wrapper,
+		title: __('Order Desk'),
+		single_column: true
+	});
+
+	wrapper.pos = new erpnext.pos.OrderDesk(wrapper);
+	window.cur_pos = wrapper.pos;
+};
+
+erpnext.pos.OrderDesk = class OrderDesk {
+	constructor(wrapper) {
+		this.wrapper = $(wrapper).find('.layout-main-section');
+		this.page = wrapper.page;
+
+		const assets = [
+			'assets/erpnext/js/pos/clusterize.js',
+			'assets/erpnext/css/pos.css'
+		];
+
+		frappe.require(assets, () => {
+			this.make();
+		});
+	}
+
+	make() {
+		return frappe.run_serially([
+			() => frappe.dom.freeze(),
+			() => {
+				this.prepare_dom();
+				this.prepare_menu();
+				this.set_online_status();
+			},
+			() => this.make_new_order(),
+			() => {
+				if(!this.frm.doc.company) {
+					this.setup_company()
+						.then((company) => {
+							this.frm.doc.company = company;
+						});
+				}
+			},
+			() => {
+				frappe.dom.unfreeze();
+			},
+		]);
+	}
+
+	set_online_status() {
+		this.connection_status = false;
+		this.page.set_indicator(__("Offline"), "grey");
+		frappe.call({
+			method: "frappe.handler.ping",
+			callback: r => {
+				if (r.message) {
+					this.connection_status = true;
+					this.page.set_indicator(__("Online"), "green");
+				}
+			}
+		});
+	}
+
+	prepare_dom() {
+		this.wrapper.append(`
+			<div class="pos">
+				<section class="cart-container">
+
+				</section>
+				<section class="item-container">
+
+				</section>
+			</div>
+		`);
+	}
+
+	make_cart() {
+		this.cart = new SalesOrderCart({
+			frm: this.frm,
+			wrapper: this.wrapper.find('.cart-container'),
+			events: {
+				on_customer_change: (customer) => {
+					this.frm.set_value('customer', customer)
+						.then((response) => {
+							this.frm.doc.items.forEach(item => {
+								this.frm.script_manager.trigger('item_code', item.doctype, item.name)
+									.then(() => {
+										this.frm.doc.taxes = [];
+										this.update_item_in_frm(item, 'qty', item.qty)
+											.then(() => {
+												// update cart
+												frappe.run_serially([
+													() => {
+													if (item.qty === 0) {
+														frappe.model.clear_doc(item.doctype, item.name);
+													}
+													},
+													() => this.update_cart_data(item),
+													() => this.post_qty_change(item)
+												]);
+											});
+									});
+							})
+						});
```

Comment:
This is not readable code. @vishdha are we doing refactor as well while moving code from Bloomstack core ?? we can break them into functions.

Reply:
yeah, most of the stuff we haven't refactored while moving to erpnext. 
Maximum Line Length
Limit all lines to a maximum of 79 characters. if we follow PEP 8

Reply:
@nehasacher143 add the inline functions inside .then statements into a separate function, that should imporve the readability.